joblib/_parallel_backends.py

```diff
@@ -265,6 +265,9 @@ def effective_n_jobs(self, n_jobs):
         This also checks if we are attempting to create a nested parallel
         loop.
         """
+        if mp is None:
```

Comment:
maybe add a comment here explaining the related problem. Something like "when multiprocessing is disabled, only one effective job is possible".
What about warning the user if n_jobs >1 and mp is None ?